src/cobo/handshake.c

```diff
@@ -410,13 +465,66 @@ static int encode_addr(struct sockaddr *addr, unsigned char *target_addr, uint16
    return 0;
 }
 
+int random_number(uint64_t *unique_value) {
+   int local_errno;
+   uint64_t rand_num, uuid_num;
+
+   //random number file
+   FILE *random_file = fopen("/dev/random", "r");
+   if (!random_file) {
+      local_errno = errno;   
+      debug_printf("Failed to open /dev/random. Error: %s\n",strerror(local_errno));
+      fclose(random_file);
+      return -1;
+   }
+   
+   if (fread(&rand_num, sizeof(rand_num), 1, random_file) != 1) {  
+      local_errno = errno;   
+      debug_printf("Failed to read random number. Error: %s\n",strerror(local_errno)); 
+      fclose(random_file);
+      return -1;
+   }
+   fclose(random_file);
+
+   //uuid file
+   FILE *uuid_file = fopen("/proc/sys/kernel/random/uuid", "r");
+   if (!uuid_file) {
+      local_errno = errno;   
+      debug_printf("Failed to open /proc/sys/kernel/random/uuid. Error: %s\n",strerror(local_errno));
+      fclose(uuid_file);
+      return -1;
+   }
+   if (fread(&uuid_num, sizeof(uuid_num), 1, uuid_file) != 1) {  
```

Comment:
A) The /proc/sys/kernel/random/uuid file returns and ASCII encoded number, but this is reading it like a binary number. 
B) If we're doing a read/fread of random data we need to handle the EINTR return codes